skops/io/_general.py

```diff
@@ -232,6 +233,25 @@ def object_get_instance(state, src):
     return instance
 
 
+def method_get_state(obj: Any, save_state: SaveState):
+    res = {
+        "__class__": obj.__class__.__name__,
+        "__module__": get_module(obj),
+        "content": {
+            "func": obj.__func__.__name__,
+            "obj": object_get_state(obj.__self__, save_state),
```

Comment:
Here you explicitly call `object_get_state`. I wonder if it would not be better to call `get_state` here. We don't know the type of the object that the method is bound to, it might need special treatment. Quickly looking at the existing special types we have, it seems unlikely that any method would be attached to them, but who knows if that might change in the future.

Reply:
Hmm, maybe. This could also be involved in the infinite recursion problem...

I'll play with this now, and double check I understand the differences between what happens when using `get_state` vs `object_get_state`

Reply:
> I'll play with this now, and double check I understand the differences between what happens when using `get_state` vs `object_get_state`

The main difference that I is when the method is bound to a type that we have a special dispatch function for. Say, the method is bound to a numpy array (which is probably never going to happen in reality). With the current implementation, `object_get_state` is hard-coded, so `ndarray_get_state` is not called for the object, resulting in a failure.